### PR TITLE
Compiler warnings and compatibility with older check versions

### PR DESCRIPTION
**Pull request by Jakuje**
While trying to build the latest version of `yubico-piv-tool` for EPEL7, I encountered several issues, mostly, the RHEL/CentOS 7 has older version of `check` (`check-0.9.9-5.el7.x86_64`), which does not provide the `ck_assert_ptr_eq`, but it can be simply rewritten to `ck_assert`. Secondly, there showed up some warnings, mostly in context of not running the HW tests at build time.

With these patches, I am able to build the package on both current Fedora and RHEL7 successfully (I did not test the HW_TEST though -- I don't have spare Yubikey at this moment).

### CODE REVIEW

File: lib/tests/api.c [hunk 1]
```diff
@@ -790,19 +790,19 @@ struct t_alloc_data{
 } g_alloc_data;
 
 static void* _test_alloc(void *data, size_t cb) {
-  ck_assert_ptr_eq(data, &g_alloc_data);
+  ck_assert(data == &g_alloc_data);
```
Comment (mrmekon) [inline lib/tests/api.c]: I'd prefer this be defined at the top like the others:

```
#ifndef ck_assert_ptr_eq
#define ck_assert_ptr_eq(a,b) ck_assert((void *)(a) == (void *)(b))
#endif
```

For libcheck 0.11+, the more specific asserts give more useful error messages.

Comment (Jakuje) [reply lib/tests/api.c]: Thank you for the comment. Yes, that was the other option. I will replace it. I did not notice these defines already on the top.

**Comment by Jakuje** [comment]
Updated based on the comment. Seems to build fine.

**Comment by mrmekon** [comment]
Cool, thanks!  I'll merge it after the Travis checks finish.

**Pull request merged**